ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/LivenessFilter.java

```diff
@@ -40,14 +40,22 @@ public LivenessFilter(final Optional<HeartbeatAgent> heartbeatAgent) {
    * @return true if the host is alive, false otherwise.
    */
   @Override
-  public boolean filter(final KsqlHostInfo host) {
+  public Host filter(final KsqlHostInfo host) {
 
     if (!heartbeatAgent.isPresent()) {
-      return true;
+      return Host.include(host);
     }
 
     final Map<KsqlHostInfo, HostStatus> allHostsStatus = heartbeatAgent.get().getHostsStatus();
     final HostStatus status = allHostsStatus.get(host);
-    return status == null ? true : allHostsStatus.get(host).isHostAlive();
+    if (status == null) {
```

Comment:
For my own education: why we treat `null` as okay alive?

Reply:
I think there was a discussion around this early on and we decided that this can happen right when a node starts up and it was better to assume alive than assume dead. I can try to dig up the history on that discussion, I believe it was on Vicky's original PR